web/source/kmwbase.ts

```diff
@@ -318,14 +318,38 @@ namespace com.keyman {
     /**
      * Exposed function to load keyboards by name. One or more arguments may be used
      *
-     * @param {string|Object} x keyboard name string or keyboard metadata JSON object
+     * @param {any[]} args keyboard name string or keyboard metadata JSON object
+     * @returns {Promise<(KeyboardStub|ErrorStub)[]} Promise of added keyboard/error stubs
      *
      */
-    ['addKeyboards'](x) {
-      if(arguments.length == 0) {
-        this.keyboardManager.keymanCloudRequest('',false);
+    async ['addKeyboards'](...args: any[]) : 
+        Promise<(com.keyman.keyboards.KeyboardStub|com.keyman.keyboards.ErrorStub)[]> {
+      if (args[0].length == 0) {
+        // Get the cloud keyboard catalog
+        let stubs: (com.keyman.keyboards.KeyboardStub|com.keyman.keyboards.ErrorStub)[] = [];
+        try {
+          let result:(com.keyman.keyboards.KeyboardStub|com.keyman.keyboards.ErrorStub)[]|Error = 
+            await this.keyboardManager.keymanCloudRequest('',false);
+          console.log('Catalog: ', result);
+          return Promise.resolve(stubs);
+        } catch(error) {
+          console.error(error);
```

Comment:
I agree with leaving new `console.error` statements in, but I'm not so sure about leaving in new `console.log` statements - those tend to feel like they were intended solely for debugging.  This one especially, since the keyboard catalog isn't exactly a small object...

Also... if `keymanCloudRequest` suffers a `reject`, does that `catch` handle the `reject` pathway?  I suspect it might; just curious.  I don't see an explicit `throw` within the function, so this is the only potential reason I see for a `catch` here.  If there is a direct error (i.e, not from a `Promise`) to catch, could you highlight where that error may arise?

Reply:
Definitely don't leave `console.log` in production code.